src/containers/Homepage/index.tsx

```diff
@@ -0,0 +1,284 @@
+import { withStyles, makeStyles } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import * as d3 from 'd3';
+import moment from 'moment';
+
+import { Padding } from './style';
+import { Title } from './style';
+import { ContentHolder } from './style';
+import { Paragraph } from './style';
+import { ImageHolder } from './style';
+import { WhiteBg } from './style';
+import { SectionTitle } from './style';
+import { Footer } from './style';
+import { ListItem } from './style';
+import { ListContainer, styles } from './style';
+import { BannerHolder } from './style';
+import { MapHolder } from './style';
+import {
+    DismissButton,
+    DialogContentWrapper,
+} from '../App/styles';
+import Banner from '../../components/Home/Banner';
+import {chartGenerator} from '../../utils/chartGenerator';
+import { loadHourlyTransactions } from '../../reducers/app/thunks';
+import {
+    getLoadingHome,
+    getHourlyTransactions
+} from '../../reducers/app/selectors';
+import { getErrorState, getMessageTxt } from '../../reducers/message/selectors';
+import { clearMessageAction } from '../../reducers/message/actions';
+
+import AskIcon from '../../assets/icons/ask_question_icon.svg';
+import AggregatedDataIcon from '../../assets/icons/aggregated_data_icon.svg';
+import ExportQueriesIcon from '../../assets/icons/export_queries_icon.svg';
+import ThoughtBubbleIcon from '../../assets/icons/thought_bubble_icon.svg';
+import PlaceholderImage from '../../assets/images/placeholder.png';
+import PlaceholderImage1 from '../../assets/images/placeholder1.png';
+import PlaceholderImage2 from '../../assets/images/placeholder2.png';
+
+import { Props, States, Transactions } from './types';
+
+class Home extends React.Component<Props, States> {
+
+    transactionPerHour: any = null;
+    axis: any = null;
+
+    constructor(props: Props) {
+        super(props);
+        
+        this.transactionPerHour = React.createRef();
+        this.axis = React.createRef();
+    }
+
+    async componentDidMount() {
+        this.fetchHourlyTransactions();
+    }
+
+    async fetchHourlyTransactions() {
+        const { loadHourlyTransactions } = this.props;
+
+        let prevDate = new Date(); 
+        // subtract one day from current time
+        prevDate.setDate(prevDate.getDate() - 1);
+
+        await loadHourlyTransactions(prevDate.getTime());
+        let {data, label, timestamps } = await this.formatHourlyTransactions(this.props.hourlytransactions, prevDate.getTime());
+
+        this.generateHourlyTransactionsGraph(data, label, timestamps);
+    }
+
+    async formatHourlyTransactions(transactionsList: Transactions[], date: number) {
+        let label: Array<Date> = [],
```

Comment:
i think can be `const`, can't see it is overwritten

Reply:
fixed